### PR TITLE
Set up Unit testing

### PR DESCRIPTION
**Pull request by WoH**

### CODE REVIEW

**Comment by johartl** [comment]
The Travis build wouldn't work for this anymore as we are testing inside the Docker container and we don't want to have a Chrome executable in that container ;)
I changed it to test outside of the container and then run `docker build` afterwards.
I also had to change some things in the `karma.conf.js`. For instance, our output folder is not `ionic2/www/build` but `server/app`. 
To be honest, I'm still not entirely sure if this test setup is correct now, especially when doing requests to the API or tests that involve navigating to different pages.

File: ionic2/karma.conf.js [hunk 1]
```diff
@@ -39,54 +30,51 @@ module.exports = function(config) {
...
-    autoWatch: true,
- 
+    autoWatch: false,
  
     // start these browsers
     // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
-    browsers: ['Chrome'],
- 
- 
+    browsers: process.env.TRAVIS ? ['ChromeTravis'] : ['Chrome'],
```
Comment (WoH) [inline ionic2/karma.conf.js:72]: Does this work? Shouldn't it be 
`browsers: process.env.TRAVIS ? [''Chrome_travis_ci''] : ['Chrome'],`

**Comment by WoH** [comment]
Missed the output part. Testing outside makes sense, I don't know about the Docker related issues unfortunately, so thanks for checking.

Navigation will be a pain by itself, but Unit testing the components should be fine.

File: ionic2/karma.conf.js (hunk 1 above)
Comment (johartl) [reply ionic2/karma.conf.js:72]: As far as I understand this refers to the `customLaunchers` property above. And it also seems to work fine:
https://travis-ci.org/PokemonGoers/Catch-em-all/builds/158177333

**Comment by WoH** [comment]
Is this good to go?

**Comment by johartl** [comment]
I just pushed two commits to replace the karma progress reporter with one that's more suitable for Travis. By that I mean one that does not print repeated lines of output like this:

```
Chromium 51.0.2704 (Ubuntu 0.0.0): Executed 0 of 1 SUCCESS (0 secs / 0 secs)
ium 51.0.2704 (Ubuntu 0.0.0): Executed 1 of 1 SUCCESS (0 secs / 0.003 secs)
ium 51.0.2704 (Ubuntu 0.0.0): Executed 1 of 1 SUCCESS (0.013 secs / 0.003 secs)
```

As far as I'm concerned you can go ahead and merge it now :)

**Comment by WoH** [comment]
Seems good to me

**Pull request merged**